pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -1243,17 +1268,53 @@ func (n *NetworkPolicyController) syncAppliedToGroup(key string) error {
 	return nil
 }
 
-// getAppliedToWorkloads returns a list of workloads like Pods and ExternalEntities matching an AppliedToGroup
+// getAppliedToWorkloads returns a list of workloads (Pods and ExternalEntities) selected by an AppliedToGroup
 // for standalone selectors or corresponding to a ClusterGroup.
 func (n *NetworkPolicyController) getAppliedToWorkloads(g *antreatypes.AppliedToGroup) ([]*v1.Pod, []*v1alpha2.ExternalEntity) {
 	// Check if an internal Group object exists corresponding to this AppliedToGroup.
-	_, found, _ := n.internalGroupStore.Get(g.Name)
+	group, found, _ := n.internalGroupStore.Get(g.Name)
 	if found {
-		return n.groupingInterface.GetEntities(clusterGroupType, g.Name)
+		// This AppliedToGroup is derived from a ClusterGroup.
+		grp := group.(*antreatypes.Group)
+		return n.getClusterGroupWorkloads(grp)
 	}
 	return n.groupingInterface.GetEntities(appliedToGroupType, g.Name)
 }
 
+// getClusterGroupWorkloads returns a list of workloads (Pods and ExternalEntities) selected by a ClusterGroup.
+// For ClusterGroup that has childGroups, the workloads are computed as the union of all its childGroup's workloads.
+// This function assumes that there's no loop in Group reference.
+func (n *NetworkPolicyController) getClusterGroupWorkloads(group *antreatypes.Group) ([]*v1.Pod, []*v1alpha2.ExternalEntity) {
+	podNameSet, eeNameSet := sets.String{}, sets.String{}
+	var pods []*v1.Pod
+	var ees []*v1alpha2.ExternalEntity
+	if len(group.ChildGroups) > 0 {
+		for _, childName := range group.ChildGroups {
```

Comment:
ditto, and it can save constucting two sets unnecessarily if processing zero childgroups case first.